src/Workspace/WorkspaceRefreshToken.js

```diff
@@ -0,0 +1,27 @@
+import { fetchWithCreds } from '../actions';
+import { wtsPath } from '../localconf';
+
+let lastRefreshMs = 0;
+const debounceMs = 60000;
+
+// start workspace session for WTS, call optional connectedCallBack if initialized/connected
+export const initWorkspaceRefreshToken = (connectedCallBack) => {
+    const nowMs = Date.now();
+    if (nowMs - lastRefreshMs > debounceMs) {
+      console.log('init/renew WTS refresh token...');
+      fetchWithCreds({
+        path: `${wtsPath}connected`,
+        method: 'GET',
+      })
+        .then(
+          ({ status }) => {
+            if (status !== 200) {
+              window.location.href = `${wtsPath}/authorization_url?redirect=${window.location.pathname}`;
```

Comment:
Notice that the redirect here only contains `window.location.pathname`, so if the page that the user is at contains some query params in its URL, they will be lost

This was not a problem before because WTS login redirect only happens in certian pages that don't need query params. But if you are moving this redirect so it can happen virtually from any portal pages, you need to accomodate accordingly, because this will have a product-wise impact

Also there are some location and history based redirect/state restore happens in other pages of portal. We should make sure this change doens't breaks them

Reply:
hi @mfshao, as discussed, I've update the redirect code to use the same logic as is used in the `src/Login/Login.jsx` page. See latest commit added to this PR: https://github.com/uc-cdis/data-portal/pull/1095/commits/8950eec53745de249fe0445cc6dffdce212fbfdf

Please let me know if this is the change you were looking for.